concourse/pipelines/deploy-cloudfoundry.yml

```diff
@@ -232,49 +198,80 @@ jobs:
           passed: ['init']
           trigger: true
         - get: bosh-CA
+        - get: cf-certs
+        - get: cf-tfstate
+          passed: ['terraform']
+      - task: terraform-variables
+        config:
+          image: docker:///ruby#2.2-slim
+          inputs:
+            - name: paas-cf
+            - name: cf-tfstate
+          run:
+            path: sh
+            args:
+            - -e
+            - -c
+            - |
+              ruby paas-cf/concourse/scripts/extract_tf_vars_from_terraform_state.rb \
+                < cf-tfstate/cf.tfstate > cf.tfvars.sh
       - task: generate-cf-certs
         config:
           image: docker:///governmentpaas/certstrap
           inputs:
           - name: bosh-CA
           - name: paas-cf
-          params:
-            CERTS_TO_GENERATE: |
-              bosh-CA:bbs_server,bbs.service.cf.internal
-              bosh-CA:bbs_client,
-              bosh-CA:router_ssl,*.${DEPLOY_ENV}.cf.paas.alphagov.co.uk
-              bosh-CA:uaa_jwt_signing,
-              bosh-CA:consul_server,server.dc1.cf.internal,server.dc2.cf.internal
-              bosh-CA:consul_agent,
-            DEPLOY_ENV: {{deploy_env}}
+          - name: cf-certs
+          - name: terraform-variables
+          outputs:
+          - name: updated-cf-certs
           run:
             path: sh
             args:
             - -e
             - -c
             - |
+              . terraform-variables/cf.tfvars.sh
+
+              CERTS_TO_GENERATE="
+              bosh-CA:bbs_server,bbs.service.cf.internal
+              bosh-CA:bbs_client,
+              bosh-CA:router_ssl,*.${TF_VAR_cf_root_domain},*.${TF_VAR_cf_apps_domain}
```

Comment:
This made me think that we should add `set -u` to this task in case the variable/output names change and we create a version containing empty strings. But that also made me think that this inline script is far too long. What do you think about moving this out to a separate script, that we'd also run shellcheck on and do away with some of the weird interpolation workarounds?